models/user_avatar.go

```diff
@@ -39,10 +39,9 @@ func (u *User) generateRandomAvatar(e Engine) error {
 	if err != nil {
 		return fmt.Errorf("RandomImage: %v", err)
 	}
-	// NOTICE for random avatar, it still uses id as avatar name, but custom avatar use md5
-	// since random image is not a user's photo, there is no security for enumable
+
 	if u.Avatar == "" {
```

Comment:
```suggestion
	if u.Avatar == "" && len(u.AvatarEmail) {
```

Reply:
well if the avatar email is 0 we probably still need to set it to something.